nilearn/image/resampling.py

```diff
@@ -445,7 +445,7 @@ def resample_img(img, target_affine=None, target_shape=None,
         if copy and not input_img_is_string:
             img = _utils.copy_img(img)
         return img
-    if target_affine is affine and target_shape is shape:
+    if target_affine is affine and target_shape == shape:
```

Comment:
```suggestion
    if (
        np.shape(target_affine) == np.shape(affine)
        and np.allclose(target_affine, affine)
        and np.array_equal(target_shape, shape)
    ):
```

Reply:
or if you are not convinced you can just revert the change to this line and we can deal with this in another pr

Reply:
It's not that I'm no convinced. I just need to invest quite some time to learn the details of this module. Also, this does go beyond the initial scope of this PR, so I think I'll follow your advice and move that commit to a new PR.

Reply:
See new PR #2925.